pkgs/games/xu4/default.nix

```diff
@@ -0,0 +1,88 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeWrapper
+, makeDesktopItem
+, zlib
+, libpng
+, boron
+, allegro5
+, libGL
+, libX11
+, libXxf86vm
+, libXcursor
+, libxml2
+, faun
+}:
+
+stdenv.mkDerivation rec {
+  pname = "xu4";
+  version = "1.3";
+
+  src = fetchFromGitHub {
+    owner = "xu4-engine";
+    repo = "u4";
+    rev = "v${version}";
+    hash = "sha256-3jyUSVfmjuKsHQKluFnQ8Gtg7aAI/WBjmY109BdNdyY=";
+    fetchSubmodules = true;
+  };
+
+  # this is not a standard Autotools-like `configure` script
+  dontAddPrefix = true;
+
+  preConfigure = ''
+    patchShebangs configure
+  '';
+
+  postConfigure = ''
+    # need to report this upstream
+    sed --in-place 's/-Wall/-Wno-error=format-security/g' src/Makefile
+
+    # disable git submodule init as part of build
+    sed --in-place 's/^\s*git submodule.*//g' src/Makefile.common
+  '';
+
+  enableParallelBuilding = true;
+
+  buildInputs = [
+    zlib
+    libpng
+    boron
+    allegro5
+    libGL
+    libX11
+    libXxf86vm
+    libXcursor
+    libxml2
+    faun
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  # there's no `install` target
```

Comment:
A make install target does exist.  It installs the dist/xu4.desktop item though, which for some reason is ignored and remade below.

Also, why does the binary get copied as $out/share/xu4?

Reply:
> A make install target does exist. 

Ah I see it's been added in 1.1 :+1: https://github.com/xu4-engine/u4/commit/bb02ea4c95c0af0534b6cf3e4112f37510e2ee28
However it includes `u4upgrade.zip` and I haven't been able to find its license so I'm packaging that separately ( https://github.com/NixOS/nixpkgs/pull/185063 ).

> It installs the dist/xu4.desktop item though, which for some reason is ignored and remade below.

Using your desktop file now as of https://github.com/NixOS/nixpkgs/pull/181866/commits/571e02812f5da12fd851557ebbacea72fc7b5121 :+1: 

> Also, why does the binary get copied as $out/share/xu4?

This is so that `bin` is clean and only contains a single binary (a small wrapper, it's pretty common in Nix) and not the modules and other files.